pkg/tls/tls.go

```diff
@@ -309,6 +324,54 @@ func getClusterIP(services corev1controllers.ServiceController) (string, error)
 	return service.Spec.ClusterIP, nil
 }
 
+func getHostIPs(deployments appscontrollers.DeploymentController, nodes corev1controllers.NodeController) ([]string, error) {
+	deployment, err := deployments.Get(namespace.System, "rancher", metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name != "rancher" {
+			continue
+		}
+		for _, port := range container.Ports {
+			if port.HostIP != "" || port.HostPort != 0 {
+				return collectNodeIPs(nodes)
+			}
+		}
+	}
+	return nil, nil
+}
+
+func collectNodeIPs(nodeController corev1controllers.NodeController) ([]string, error) {
+	nodes, err := nodeController.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	nodeIPs := make([]string, 0, len(nodes.Items))
+	for _, node := range nodes.Items {
+		var nodeIP string
+		for _, ip := range node.Status.Addresses {
```

Comment:
Is the idea here that we are trying to prefer using the internal node IP? It looks like we take preference to only capturing the internal node IP but I don't think that is fully valid, as the external IP can be used to hit the node as well.